test/db/index.js

```diff
@@ -0,0 +1,1539 @@
+        expect(event).excluding(['_id']).to.deep.not.equal(updateNewEvent);
+        await DBHelper.updateEvent(mockEvent[2].txid, updateNewEvent);
+        eventCount = await db.Events.count({});
+        eventCount.should.equal(3);
+        event = await DBHelper.findOneEvent({ txid: updateNewEvent.txid });
+        should.exist(event);
+        expect(event).excluding(['_id']).to.deep.equal(updateNewEvent);
+      });
+
+      it('update event by address', async () => {
+        let event = await DBHelper.findOneEvent({ address: '0x0c97ba43149c1232920ef1e9d410316905c1ce8a' });
+        expect(event).excluding(['_id']).to.deep.not.equal(mockEvent[2]);
+        await DBHelper.updateEventByAddress('0x0c97ba43149c1232920ef1e9d410316905c1ce8a', mockEvent[2]);
+        eventCount = await db.Events.count({});
+        eventCount.should.equal(3);
+        event = await DBHelper.findOneEvent({ txid: mockEvent[2].txid });
+        should.exist(event);
+        expect(event).excluding(['_id']).to.deep.equal(mockEvent[2]);
+      });
+    });
+
+    describe('event status update', () => {
+      let event;
+
+      describe('update event status to PRE_BETTING', () => {
+        const mockCreatedStatusEvent = {
+          txType: 'CREATE_EVENT',
+          txid: '0x1628812210d8e7d071ee343ece4d79c248191fdf76938cecd572158c475cb7f5',
+          txStatus: 'SUCCESS',
+          blockNum: 4404133,
+          address: '0x1c97ba43149c6eee920ef1e9d410316905c1ce8a',
+          ownerAddress: '0x47ba776b3ed5d514d3e206ffee72fa483baffa7e',
+          version: 6,
+          name: 'TEST EVENT STATUS',
+          results: [
+            'Invalid',
+            '1',
+            '2',
+          ],
+          numOfResults: 3,
+          centralizedOracle: '0x47ba776b3ed5d514d3e206ffee72fa483baffa7e',
+          betStartTime: 1561585230,
+          betEndTime: 1561671614,
+          resultSetStartTime: 1561671714,
+          resultSetEndTime: 1561844414,
+          escrowAmount: '100000000',
+          arbitrationLength: 86400,
+          thresholdPercentIncrease: '10',
+          arbitrationRewardPercentage: 10,
+          currentRound: 0,
+          currentResultIndex: 255,
+          consensusThreshold: '10000000000',
+          arbitrationEndTime: 0,
+          status: 'CREATED',
+          language: 'en-US',
+          withdrawnList: [
+          ],
+        };
+        it('should not update event status to PRE_BETTING if txStatus is FAIL', async () => {
+          const createdEvent = {};
+          Object.assign(createdEvent, mockCreatedStatusEvent, { txStatus: TX_STATUS.FAIL, address: padHex(40), txid: padHex(64) });
+          await DBHelper.insertEvent(createdEvent);
+          await DBHelper.updateEventStatusPreBetting(createdEvent.betStartTime - 1);
+          event = await DBHelper.findEvent({ address: createdEvent.address });
+          event.length.should.equal(1);
+          expect(event[0]).excluding(['_id']).to.deep.equal(createdEvent);
```

Comment:
dont need to check every field for this test. you wanna check the event `status` and make sure it stays `CREATED`.

instead of checking every field, check `txid` or `address` to verify that it is the event you inserted. and also check the `status` did not change from `CREATED`.

i know your check does handle it, but you want to be focused when testing a specific case. like for example, if some change broke another part of the event, say like the name or something. then this test would fail cause of that as well. yes it does make test writing more tedious, but it serves a purpose.

Reply:
please go through the all the event status tests and adjust accordingly